drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -349,6 +353,42 @@ int intel_adsp_hda_dma_stop(const struct device *dev, uint32_t channel)
 		return 0;
 	}
 
+	/*
+	 * Workaround for random metallic noise on DMA restart:
+	 * Make sure that the output link DMA read and write pointers are exactly 0 when we disable
+	 * the channel.
+	 * On channel restart the pointers are not rested to 0 and can result corruption.
+	 */
+	if (cfg->direction == MEMORY_TO_PERIPHERAL) {
```

Comment:
@ujfalusi do you think it might be better to add a  kconfig option like DMA_DRAIN_STOP or something along those lines and make this opt in?

Reply:
@ranj063, yes, that might be a good idea, also I would lift the playback only restriction.